lib/index.js

```diff
@@ -2850,12 +2858,16 @@ addToplevelMethod('pipeline', function (/*through...*/) {
         });
     });
 
+    wrapper.writable = true;
+
     wrapper.write = function (x) {
         return start.write(x);
     };
 
     wrapper.end = function () {
-        return start.end();
+        var result = start.end();
+        wrapper._onEnd();
```

Comment:
I don't think I understand that last comment. Do you mean in a new else clause on the if on line 2855? Or add an event listener to something?